### ISSUE TITLE
Add support to access parent block data from child blocks

### ISSUE THREAD

**Issue by Resolution-Sam** [WordPress/gutenberg#43159]
## What problem does this address?
I need to access data from a parent block so it can be shared with the nested child blocks. E.g. a class name, data attribute etc... 

One use case I stumbled across is building an accordion with native Gutenberg blocks. For accessibility reasons, I need a shared data attribute ID between my core/heading and core/paragraph block. I can generate the ID on the fly but it's not possible to generate the same ID for both the heading and paragraph block. If the ID was stored on the parent block I could then access and share this data with the child blocks.

## What is your proposed solution?
Using the `add_filter('render_block')` hook allow access to parent block data.

**Comment by Mamaduka** [comment]
Hi, @Resolution-Sam

I would recommend checking the developer documentation about Block Context API - https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/.

**Comment by tellthemachines** [comment]
@Resolution-Sam did the above answer your question? If so it would be good to close the issue now.

**Comment by tellthemachines** [comment]
Closing this issue due to lack of response from OP.

**Comment by fukou** [comment]
I sincerely apologize for reopening the discussion on the closed ticket.

I've encountered a similar situation where I had to access data from a parent block to a nested child block. I have utilized the Block Context API for the custom block I'm currently developing, as per the suggestion and it's working!

However, according to the documentation, the Block Context is not available on the `Save` block and only works on the `Edit` block.

Is there a way to achieve this functionality during the `Save` blocks as well? Despite searching across various platforms, I haven't found a suitable solution.

Any guidance in the right direction would be greatly appreciated! 
Thank you 🙏
